octobot_trading/exchange_data/prices/price_events_manager.py

```diff
@@ -75,6 +77,23 @@ def add_event(self, price, timestamp, trigger_above):
         self.events.append(price_event_tuple)
         return price_event_tuple[PriceEventsManager.PRICE_EVENT_INDEX]
 
+    def is_triggered_by_last_recent_trades(self, price, timestamp, trigger_above):
```

Comment:
I think we shouldn't trigger this manually, we should check each time we try to create a new price event. It should be abstracted for the price event client.

Reply:
Currently when adding an event, we create this event and return it. We could check if this event would be filled instantly when creating it and if so, set it instantly. Then the client would have to check if the event is set or not before creating a waiting task or proceeding as now when instantly filled.
What do you think about this approach ?

Reply:
it's a good question. I think we need to abstract this internal mechanism because, in my opinion, a client shouldn't have to check before asking. A client just wants to "wait for price hit". If we set the event instantly the client has to check before waiting that the event is not set right? I think it's better, it seems logical to me